src/vdaf/hits.rs

```diff
@@ -0,0 +1,741 @@
+        let mut data = vec![F::zero(); data_len];
+        let mut output = output.into_iter();
+        for len in 0..input.len + 1 {
+            data[input.prefix(len).index()] = output.next().unwrap();
+        }
+
+        // NOTE(cjpatton) We could save some space by representing one of the key shares as a PRNG
+        // seed. Of course, there's not much point, since this is a toy.
+        let mut data = split_vector(&data, 2)?.into_iter();
+        Ok([
+            ToyIdpf {
+                data: data.next().unwrap(),
+                len: input.len,
+            },
+            ToyIdpf {
+                data: data.next().unwrap(),
+                len: input.len,
+            },
+        ])
+    }
+
+    fn eval(&self, prefix: &IdpfInput) -> Result<[F; 1], VdafError> {
+        if prefix.len > self.len {
+            return Err(VdafError::Uncategorized(format!(
+                "prefix length ({}) exceeds input length ({})",
+                prefix.len, self.len
+            )));
+        }
+
+        Ok([self.data[prefix.index()]])
+    }
+}
+
+/// An input share for the heavy hitters VDAF.
+pub struct InputShareMessage<I: Idpf<2, 1>> {
+    /// IDPF share of inpuot
+    pub data: I,
+
+    /// IDPF share of the authentication vector
+    pub auth: I,
+
+    /// PRNG seed used to generate the aggregator's share of the randomness used in the first part
+    /// of the sketching protocol.
+    pub sketch_start_seed: Key,
+
+    /// Aggregator's share of the randomness used in the second part of the sketching protocol.
+    pub sketch_next: Share<I::Field>,
+}
+
+/// The VDAF input-distribution algorithm for heavy hitters.
+#[allow(clippy::many_single_char_names)]
+pub fn hits_input<I: Idpf<2, 1>>(
+    suite: Suite,
+    input: &IdpfInput,
+) -> Result<[InputShareMessage<I>; 2], VdafError> {
+    let auth_rand: Vec<I::Field> = Prng::generate(suite)?.take(input.len + 1).collect();
+
+    // For each level of the prefix tree, generate correlated randomness that the aggregators use
+    // to validate the output. See [BBCG+21, Appendix C.4].
+    let leader_sketch_start_seed = Key::generate(suite)?;
+    let helper_sketch_start_seed = Key::generate(suite)?;
+    let helper_sketch_next_seed = Key::generate(suite)?;
+    let mut leader_sketch_start_prng: Prng<I::Field> =
+        Prng::from_key_stream(KeyStream::from_key(&leader_sketch_start_seed));
+    let mut helper_sketch_start_prng: Prng<I::Field> =
+        Prng::from_key_stream(KeyStream::from_key(&helper_sketch_start_seed));
+    let mut helper_sketch_next_prng: Prng<I::Field> =
+        Prng::from_key_stream(KeyStream::from_key(&helper_sketch_next_seed));
+    let mut leader_sketch_next: Vec<I::Field> = Vec::with_capacity(2 * input.len);
+    for k in auth_rand.iter() {
+        // [BBCG+21, Appendix C.4]
+        //
+        // $(a, b, c)$
+        let a = leader_sketch_start_prng.next().unwrap() + helper_sketch_start_prng.next().unwrap();
```

Comment:
This is a lot of `unwrap()`. Since we are always dealing with a concrete instance of `Prng` rather than generically using some `Iterator`, maybe we should put a method on `Prng` that infallibly returns a value.

Reply:
I'll do this in a PR after this one to avoid touching loads more files in this change.